drivers/usb/device/usb_dc_sam0.c

```diff
@@ -262,22 +272,25 @@ int usb_dc_attach(void)
 int usb_dc_detach(void)
 {
 	UsbDevice *regs = &REGS->DEVICE;
+	struct usb_sam0_data *data = usb_sam0_get_data();
 
 	regs->CTRLB.bit.DETACH = 1;
 	usb_sam0_wait_syncbusy();
-
+	data->cb(USB_DC_DISCONNECTED, NULL);
```

Comment:
Same with this. This should eventually lead to `SUSPEND` interrupt getting triggered.

`USB_DC_SUSPEND` seems to do all the necessary clean-ups and AFAICS there is no way to truly detect "detach"/"disconnect" event in the SAM0 USB device core, so `USB_DC_DISCONNECTED` should be left unimplemented (IIUC, this is optional).